api/src/main/java/io/minio/ServerSideEncryption.java

```diff
@@ -37,209 +38,203 @@
+  /**
+   * Returns the type of server-side encryption.
+   * @return the type of server-side encryption.
+   *
+   * @deprecated As of release 7.0
+   */
+  @Deprecated
+  public Type getType() {
+    return this.type();
   }
 
- /**
-  * Create a new server-side-encryption object for encryption at rest (a.k.a. SSE-S3).
-  *
-  * @return an instance of ServerSideEncryption implementing SSE-S3  
-  */
-  public static ServerSideEncryption atRest() {
-    return new ServerSideEncryptionS3(); 
+  /**
+   * Set the server-side-encryption headers of this specific encryption.
+   * @param headers The metadata key-value map.
+   *
+   * @deprecated As of release 7.0
+   */
+  @Deprecated
+  public void marshal(Map<String, String> headers) {
+    headers.putAll(this.headers());
   }
 
- /**
-  * Create a new server-side-encryption object for encryption using a KMS (a.k.a. SSE-KMS).
-  * 
-  * @param keyId   specifies the customer-master-key (CMK) and must not be null.
-  * @param context is the encryption context. If the context is null no context is used.
-  *
-  * @return an instance of ServerSideEncryption implementing SSE-KMS.  
-  */
-  public static ServerSideEncryption withManagedKeys(String keyId, Map<String,String> context) 
-    throws InvalidArgumentException, UnsupportedEncodingException {
-    if (keyId == null) {
-      throw new InvalidArgumentException("The key-ID cannot be null");
-    }
-    if (context == null) {
-      return new ServerSideEncryptionKms(keyId, Optional.empty());
-    }
+  @Override
+  public boolean isDestroyed() {
+    return this.destroyed;
+  }
 
-    StringBuilder builder = new StringBuilder();
-    int i = 0;
-    builder.append('{');
-    for (Entry<String,String> entry : context.entrySet()) {
-      builder.append('"');
-      builder.append(entry.getKey());
-      builder.append('"');
-      builder.append(':');
-      builder.append('"');
-      builder.append(entry.getValue());
-      builder.append('"');
-      if (i < context.entrySet().size() - 1) {
-        builder.append(',');
-      }
+  private static boolean isCustomerKeyValid(SecretKey key) {
+    if (key == null) {
+      return false;
     }
-    builder.append('}');
-    String contextString = BaseEncoding.base64().encode(builder.toString().getBytes("UTF-8"));
-    return new ServerSideEncryptionKms(keyId, Optional.of(contextString)); 
+    return !key.isDestroyed() && key.getAlgorithm().equals("AES") && key.getEncoded().length == 32;
   }
 
   static class ServerSideEncryptionWithCustomerKey extends ServerSideEncryption {
-
     protected final SecretKey secretKey;
-    protected final MessageDigest md5;     
-   
+    protected final MessageDigest md5;
+
     public ServerSideEncryptionWithCustomerKey(SecretKey key, MessageDigest md5) {
       this.secretKey = key;
       this.md5 = md5;
     }
 
     @Override
-    public final Type getType() { 
+    public final Type type() { 
```

Comment:
Java uses getter / setter for properties. Renaming is not idiomatic here.

Reply:
It is old style and minio-java follows this style inspired from okhttp.

Reply:
Okay, can/have we document that? Because a **lot** of Java code follows the "style" I mentioned...

Reply:
There is no coding standard document we have. Modern style is followed from the beginning of minio-java. There is little different with getter style where getters may return reference of an class member but this style returns copy of a value.

Reply:
Okay